### ISSUE TITLE
What is meaning of 121 in the notation DenseNet-121?

### ISSUE THREAD

**Issue by mjohn123** [liuzhuang13/DenseNetCaffe#9]
Thank you for sharing nice work!
This is not bug. I just want to clarify my points
1. In your Table 1, you used `DenseNet-121`, `DenseNet-169`...What does it means 121? How to compute it? If it is the depth of network, what is relationship with `L` term?
2. In your solver.prototxt, why do you use so big learning rate? They often use very small learning rate (for Adam method) like 0.001, instead of 0.1. The reason is you used other method (Nesterov method), so you can you very high learning rate=0.1. Is it right?

**Update:** This is my solver using Adam method

```
train_net: "train_densenet_BC.prototxt"
display: 20
lr_policy: "step"
gamma: 0.1
stepsize: 20000
power: 0.75
# lr for normalized softmax
base_lr: 0.001
# high momentum
momentum: 0.99
# no gradient accumulation
iter_size: 1
max_iter: 100000
weight_decay: 0.0005
snapshot: 4000
snapshot_prefix: "snapshot/train_dense"
type:"Adam"
```

**Comment by liuzhuang13** [comment]
Hi,

1. 121 and 169 denotes the depth of the ImageNet models. "L" is also the depth, in the paper I believe it's mainly used for CIFAR models.
2. We follow the learning rate schedule of ResNet, it seems to work pretty well. To be honest, I don't know the reason why they use such a large learning rate

Thanks

**Comment by mjohn123** [comment]
Thank you for your information. 
1. In your implementation, you used `deep=L=3n+4`. I undertood that `3` means the number of layer in each dense block. These layers are (BN+Scale+ReLU), (Conv) and (Dropout). However, what is `4`? I guess is is 1conv+ 2 transition layer+1 pooling. But the number of transition layer will be grow up as L increasing. So `4` will not be constant.
If we extends to ImageNet, which uses 4 layers in each dense block (BN+Scale+ReLU), (Conv 1x1), (BN+Scale+ReLU), (Conv) . So the equation will be `deep=L=4n+x`. What is x?

Thanks in advance?

**Comment by liuzhuang13** [comment]
In CIFAR implementation (non-BC structure), L = 3n+4, where 3 means number of dense blocks, n means number of conv layers in each dense block, 4 means 1 conv (before 1st dense block) + 2 transition + 1 classifcation).

In ImageNet implementation, if n means number of (1x1 + 3x3) conv layer compositions, then L = 4n + 5, because we have 4 dense blocks thus 3 transition layers here. 

In general, only conv and classification layers count as depth, we don't count BN, dropout, ReLU as depth in our paper or code.

**Comment by mjohn123** [comment]
Great. I understood. 
You said that "Our ImageNet model didn't use 4 layers in each dense block.". Do you use 2 layers in each dense block [conv 1x1, conv 3x3]? 
I agree that we can count `conv` and `classification` only, because `BN, ReLU` can be used as `in_place` to saving memory. Regarding ImageNet model, if my depth is less than 50. Do you think that I should use the configuration of the layer in the dense block as [conv 3x3, BN-ReLU, conv 3x3 ], instead of your bottleneck layer [conv 1x1, BN-ReLU, conv 3x3] as suggestion in ResNet?

**Comment by liuzhuang13** [comment]
Check our paper for the definition of "dense block", it doens't mean one or two convolution transformations. 

I think conv 1x1 + 3x3 works well, we didn't try conv 3x3 + 3x3, that's two many parameters.

**Comment by mjohn123** [comment]
Thanks for the valuable discussion. I understood it. Currently, I am using your network for my database. I have not so much training data. My network depth is about 50 layers with 4 dense blocks. I'm wondering between BC and non-BC structure. With BC structure and reduction =50%, my network has about 2MB parameters, while non-BC structure has 11MB parameters (so much). My aim is a low error rate. Table2 shows that with depth is less than 100, non-BC looks better. Could you suggest to me which model is better for my case?

**Comment by liuzhuang13** [comment]
BC structure is better in terms of parameter efficiency. A 2MB parameter BC model will probably be less accurate than a 11MB non-BC model, but if you make your BC model to be of comparable size with the non-BC one, I think probably the BC one will do better. Good luck!
